pkg/apis/core/validation/validation.go

```diff
@@ -902,6 +902,23 @@ func validateGlusterfsVolumeSource(glusterfs *core.GlusterfsVolumeSource, fldPat
 	}
 	return allErrs
 }
+func validateGlusterfsPersistentVolumeSource(glusterfs *core.GlusterfsPersistentVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(glusterfs.EndpointsName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("endpoints"), ""))
+	}
+	if len(glusterfs.Path) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
+	}
+	if glusterfs.EndpointsNamespace != nil {
+		endpointNs := glusterfs.EndpointsNamespace
+		if len(*endpointNs) == 0 {
+			allErrs = append(allErrs, field.Required(fldPath.Child("endpointsNamespace"), ""))
+		}
+		allErrs = append(allErrs, ValidateDNS1123Label(*endpointNs, fldPath.Child("endpointsNamespace"))...)
```

Comment:
don't run the format check if the length is 0 and we already added a "required" error